app/models/concerns/orchestration/dhcp.rb

```diff
@@ -19,6 +19,11 @@ def dhcp?
         !subnet.nil? && subnet.dhcp? && SETTINGS[:unattended] && (!provision? || operatingsystem.present?)
   end
 
+  def generate_id(action, interface = self)
```

Comment:
Can we name this `generate_dhcp_task_id`? This is a concern so every Nic will have the method `generate_id` which is a bit confusing imho. Otherwise: Looks good.

Reply:
Sure good point.